geometric_kernels/spaces/hypersphere.py

```diff
@@ -0,0 +1,188 @@
+"""
+Spaces for which there exist analytical expressions for the manifold
+and the eigenvalues and functions. Examples include the `Circle` and the `Hypersphere`.
+The Geomstats package is used for most of the geometric calculations.
+"""
+
+from typing import Tuple
+
+import geomstats as gs
+import lab as B
+from spherical_harmonics import SphericalHarmonics as _SphericalHarmonics
+from spherical_harmonics.fundamental_set import num_harmonics
+
+from geometric_kernels.eigenfunctions import (
+    Eigenfunctions,
+    EigenfunctionWithAdditionTheorem,
+)
+from geometric_kernels.spaces import DiscreteSpectrumSpace
+from geometric_kernels.utils import Optional, chain
+
+
+class SphericalHarmonics(EigenfunctionWithAdditionTheorem):
+    """
+    Eigenfunctions Laplace-Beltrami operator on the sphere correspond to the
+    spherical harmonics.
+    """
+
+    def __init__(self, dim: int, num_eigenfunctions: int) -> None:
+        """
+        :param dim:
+            S^{dim}. Example: dim = 2 is the sphere in R^3. Follows geomstats convention.
+
+        :param num_eigenfunctions:
+            Specifies the minimum degree of the spherical harmonic collection so that
+            there are at least `num_eigenfunctions`.
+        """
+        self.dim = dim
+        self._num_levels, self._num_eigenfunctions = self.num_eigenfunctions_to_degree(
+            num_eigenfunctions
+        )
+        self._spherical_harmonics = _SphericalHarmonics(
+            dimension=dim + 1, degrees=self._num_levels
+        )
+
+    def num_eigenfunctions_to_degree(self, num_eigenfunctions: int) -> Tuple[int, int]:
+        """
+        Returns the minimum degree for which there are at least
+        `num_eigenfunctions` in the collection.
+        """
+        n, d = 0, 0
```

Comment:
This is maybe a naive question/comment, but could we comment on what `n` and `d` are? This may be useful for people that are not very familiar with spherical harmonics.